packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,697 @@
+		      "message": "Invalid ExperimentAssignment",
+		    },
+		  ],
+		  Array [
+		    Object {
+		      "experimentName": "experiment_name_a",
+		      "message": "Invalid ExperimentAssignment",
+		    },
+		  ],
+		  Array [
+		    Object {
+		      "experimentName": "experiment_name_a",
+		      "message": "Invalid ExperimentAssignment",
+		    },
+		  ],
+		]
+	` );
+
+		await delayedValue( undefined, 1000 );
+
+		jest.resetAllMocks();
+		const dateIncreasePastTtl = validExperimentAssignment.ttl * 1000 + 10;
+		const refreshedValidExperimentAssignment = {
+			...validExperimentAssignment,
+			variationName: null,
+			retrievedTimestamp: Date.now() + dateIncreasePastTtl,
+		};
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( refreshedValidExperimentAssignment );
+		spiedMonotonicNow.mockImplementation( () => Date.now() + dateIncreasePastTtl );
+		const experimentAssignmentE = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		await experimentAssignmentE;
+		const experimentAssignmentF = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentG = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentH = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		expect( await experimentAssignmentE ).toEqual( await experimentAssignmentF );
+		expect( await experimentAssignmentE ).toEqual( await experimentAssignmentG );
+		expect( await experimentAssignmentE ).toEqual( await experimentAssignmentH );
+		expect( ( await experimentAssignmentA ).retrievedTimestamp ).toBeLessThan(
+			( await experimentAssignmentE ).retrievedTimestamp
+		);
+		expect( mockedFetchExperimentAssignment.mock.calls ).toHaveLength( 1 );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+	it( `[developmentMode] should only make one request even if it fails, throwing the same error`, async () => {
+		const client = createExPlatClient( createMockedConfig( DEVELOPMENT_MODE ) );
+
+		const invalidExperimentAssignment = {
+			...validExperimentAssignment,
+			experimentName: 'invalid-experiment-name',
+		};
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( invalidExperimentAssignment );
+		const firstDate = Date.now();
+		spiedMonotonicNow.mockImplementation( () => firstDate );
+		const experimentAssignmentA = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentB = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentC = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		const experimentAssignmentD = client.loadExperimentAssignment(
+			validExperimentAssignment.experimentName
+		);
+		let errorA;
+		try {
+			await experimentAssignmentA;
+		} catch ( e ) {
+			errorA = e;
+		}
```

Comment:
It'd be nice to add an `awaitException()` helper that'd return the error. Then we can do `const errorA = awaitException( experimentAssignmentA )`. The helper can also fail if the exception isn't thrown.